### PR TITLE
initial change for uvs retrieval 2024-03-01-preview

### LINKED ISSUE #27498: [Unified Vision Service ] API Review
New API Review meeting has been requested.

**Service Name**: Unified Vision Service 
**Review Created By**: Kailash Joshi
**Review Date**: 02/15/2024 04:00 PM PT
**Onboarding Record**: 
**PR**: 
**Hero Scenarios Link**: Not Provided
**Core Concepts Doc Link**: Not Provided

**Description**: Search/Ingestion API for computer vision

Detailed meeting information and documents provided can be accessed [here](https://web.powerapps.com/apps/e41c69de-ef96-44ba-b287-0e012e7c13ae?view=edit&event=c3fb5a8d-16ba-ee11-a569-000d3a3411c3)
For more information that will help prepare you for this review, the requirements, and office hours, visit the documentation [here](https://aka.ms/azsdk/onboarding/restapischedule)

### PR DESCRIPTION
**Pull request by MinjingZhu**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*API Spec engagement record*](https://aka.ms/azsdk/onboarding/restapischedule).

* Link to API Spec engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope

<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous API Spec document (if applicable), and the root paths that have been updated. </sup>
* Design Document:
* Previous API Spec Doc:
* Updated paths:

### Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

### Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/pr-suppressions) 
to get approval.

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azsdk/onboarding/restapischedule) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azsdk/onboarding/restapischedule).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling

 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://github.com/Azure/azure-api-style-guide/blob/main/README.md)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications

 * [Azure REST API Guidelines](https://aka.ms/azapi/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/azapi/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links

 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

<details>
  <summary>Checks stuck in `queued` state?</summary>
If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.  
</details>

### CODE REVIEW

File: specification/cognitiveservices/data-plane/UnifiedVision/search/readme.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+``` yaml
+
+tag: 2024-03-01-preview
+openapi-type: data-plane
+add-credentials: true
+```
+## Releases
+
+### 2023-10-01
```
Comment (Kai-lash) [inline specification/cognitiveservices/data-plane/UnifiedVision/search/readme.md:15]: Please update this to the new version.

File: specification/cognitiveservices/data-plane/UnifiedVision/search/preview/2024-03-01-preview/UnifiedVision-Search.json [hunk 2]
```diff
@@ -0,0 +1,1227 @@
...
+        }
+      }
+    },
+    "/retrieval/indexes": {
+      "get": {
+        "tags": [
+          "IngestionApi"
+        ],
+        "summary": "Retrieves all indexes.",
+        "description": "Retrieves a list of all indexes across all ingestions.",
```
Comment (Kai-lash) [inline specification/cognitiveservices/data-plane/UnifiedVision/search/preview/2024-03-01-preview/UnifiedVision-Search.json:191]: As you can see, one of the checks is spell check. It is complaining about the word 'ingestions'. Please put it and any other words you see in the spell check in to the custom-words.txt file present at the root of this repo.

File: specification/cognitiveservices/data-plane/UnifiedVision/search/readme.md (hunk 1 above)
Comment (MinjingZhu) [reply specification/cognitiveservices/data-plane/UnifiedVision/search/readme.md]: updated, thanks!

File: specification/cognitiveservices/data-plane/UnifiedVision/search/preview/2024-03-01-preview/UnifiedVision-Search.json (hunk 2 above)
Comment (MinjingZhu) [reply specification/cognitiveservices/data-plane/UnifiedVision/search/preview/2024-03-01-preview/UnifiedVision-Search.json:191]: added

File: specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json [hunk 3]
```diff
@@ -0,0 +1,1289 @@
...
+            "minimum": 0
+          },
+          {
+            "in": "query",
+            "name": "top",
+            "description": "Number of datasets to be returned after skipping.",
+            "type": "integer",
+            "format": "int32",
+            "default": 10,
+            "maximum": 30,
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json:472]: No default or maximum on `top` parameter.

File: specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json [hunk 4]
```diff
@@ -0,0 +1,1289 @@
...
+    "/retrieval/indexes/{indexName}:queryByText": {
+      "post": {
+        "tags": [
+          "SearchApi"
+        ],
+        "summary": "Performs a text-based search.",
+        "description": "Performs a text-based search on the specified index.",
+        "operationId": "SearchByText",
+        "consumes": [
+          "application/json-patch+json"
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json:512]:
```suggestion
          "application/json"
```

File: specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json [hunk 5]
```diff
@@ -0,0 +1,1289 @@
...
+    },
+    "IngestionResponse": {
+      "description": "Represents the ingestion response model for the JSON document.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Gets or sets the name of the ingestion.",
+          "type": "string"
+        },
+        "status": {
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json:953]: `status` property in LRO response should be required

File: specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json [hunk 6]
```diff
@@ -0,0 +1,1289 @@
...
+        },
+        "innerError": {
+          "$ref": "#/definitions/IngestionInnerErrorDetails"
+        }
+      }
+    },
+    "IngestionResponse": {
+      "description": "Represents the ingestion response model for the JSON document.",
+      "type": "object",
+      "properties": {
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json:948]: LRO response should contain top-level property `id`

File: specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json [hunk 7]
```diff
@@ -0,0 +1,1289 @@
...
+        "name": {
+          "description": "Gets or sets the name of the ingestion.",
+          "type": "string"
+        },
+        "status": {
+          "description": "Gets or sets the status of the ingestion.",
+          "enum": [
```
Comment (mikekistler) [inline specification/cognitiveservices/data-plane/UnifedVisionRetrieval/preview/2024-03-01-preview/UnifiedVisionRetrieval.json:955]: 'status' property enum in LRO response should contain values: Running, Succeeded, Failed, Canceled

**Review by mikekistler — CHANGES_REQUESTED** [review]
Here's another round of issues to be addressed.

**Pull request closed without merging**